### ISSUE TITLE
File editing should be available without connecting to a server

### ISSUE THREAD

**Issue by gyorokpeter** [ScottLogic/kdb-studio-2#71]

**Comment by ColinEberhardt** [comment]
This leads to a broader question is Boothroyd either:

1. A database GUI with a built-in scrip editor
2. A script editor with the ability to connect to a database

We are opting for (1), hence features such as a table explorer. This is similar to tools such as Squirrel SQL.

**Comment by gyorokpeter** [comment]
I don't think this is a matter of choice if we are aiming to replace KDB Studio. We need to follow its overall design principles, which includes that the server connection only needs to be alive if you actually want to do any queries. Many users have loads of different servers with no good choice for one to use as the "default". Also these servers may keep coming up and down. So locking the editor behind having to open a server connection would be a hindrance in terms of UX.
